### PR TITLE
Upgrade kind, kind images and add testing for Kubernetes 1.25

### PR DESCRIPTION
**Pull request by iblancasa**
Signed-off-by: Israel Blancas <iblancasa@gmail.com>

## Which problem is this PR solving?
- Upgrade `kind` version used
- Upgrade `kind` container images used
- Add Kubernetes 1.25 testing
- Make the E2E tests compatible with the `BatchV1/Cronjobs` API (previously, only supported `BatchV1beta1/Cronjobs`)
- Upgrade the Kafka Operator version used for the E2E Tests from `0.23.0` to `0.30.0`. `0.23.0` is not compatible with Kubernetes 1.25
- Upgrade the Istio version from `0.11.2` to `0.15.0`. `0.11.2` is not compatible with Kubernetes 1.25.

### CODE REVIEW

**Comment by frzifus** [comment]
@iblancasa a few tests failed. Could you have a look?

**Comment by iblancasa** [comment]
I know there still some stuff failing. I'm trying to fix it step by step. This release broke some stuff

**Comment by iblancasa** [comment]
@frzifus it seems everything is working right now :) Could you take a look?

Thanks!

File: tests/cmd-utils/wait-cronjob/main.go [hunk 1]
```diff
@@ -43,25 +43,57 @@ func checkCronJobExists(clientset *kubernetes.Clientset) error {
 		ctxWithTimeout, cancel := context.WithTimeout(context.Background(), timeout)
 		defer cancel()
 
-		cronjobs, err := clientset.BatchV1beta1().CronJobs(namespace).List(ctxWithTimeout, metav1.ListOptions{})
+		beta1v1JobsFound := false
+
+		cronjobsV1beta1, err := clientset.BatchV1beta1().CronJobs(namespace).List(ctxWithTimeout, metav1.ListOptions{})
+		if err == nil {
+			beta1v1JobsFound = true
+		} else {
```
Comment (frzifus) [inline tests/cmd-utils/wait-cronjob/main.go:51]: do we need that else? What about something like this?
```golang
beta1v1JobsFound := true
cronjobsV1beta1, err := clientset.BatchV1beta1().CronJobs(namespace).List(ctxWithTimeout, metav1.ListOptions{})
if err != nil  {
  beta1v1JobsFound = false
  if apierrors.IsNotFound(err) {
    logrus.Debug("No BatchV1beta1/Cronjobs were found")
  }
}
```

**Review by frzifus — COMMENTED** [review]
LGTM, but iam not able to merge it, since 1.24 is pending. @rubenvp8510 could you have a look?

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**